lib/src/main/java/com/dynatrace/metric/util/NumberValueValidator.java

```diff
@@ -59,9 +60,49 @@ static BooleanResultMessage isSummaryValid(double min, double max, double sum, l
               String.format(
                   ValidationMessages.GAUGE_MIN_GREATER_MAX_MESSAGE, min, max, sum, count));
     }
+
+    double avg = sum / count;
+    if (!(lessOrEqualWithAbsoluteTolerance(min, avg)
+        && lessOrEqualWithAbsoluteTolerance(avg, max))) {
+      // in this case the min <= avg <= max does not hold
+      return BooleanResultMessage.newInvalid(
+          () ->
+              String.format(
+                  ValidationMessages.GAUGE_INCONSISTENT_FIELDS_MESSAGE,
+                  min,
+                  max,
+                  sum,
+                  count,
+                  avg,
+                  COMPARISON_ABSOLUTE_TOLERANCE));
+    }
+
     return BooleanResultMessage.newValid();
   }
 
+  private static boolean lessOrEqualWithAbsoluteTolerance(double val1, double val2) {
+    // equals
+    if (Double.valueOf(val1).equals(Double.valueOf(val2))) {
+      return true;
+    }
+
+    // not equal, two cases:
+    // val1 < val2: should always evaluate to true
+    // val1 slightly < val2 -> val1 - val2 == negative     | result <= tolerance -> true
+    // val1 much < val2 -> val1 - val2 == negative         | result <= tolerance -> true
+    //
+    // val2 > val2: should only evaluate to true if
```

Comment:
```suggestion
    // val1 > val2: should only evaluate to true if
```
Right?

Reply:
Yes, you are right. The line below did not make sense either, see 4dbcba9